sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -534,4 +524,22 @@ int getTotalRecordedLinks() {
         ? Collections.unmodifiableMap(attributes)
         : Collections.unmodifiableMap(new HashMap<>(attributes));
   }
+
+  private static class LimitingAttributeConsumer implements KeyValueConsumer<AttributeValue> {
+    private final int limit;
+    private final Attributes.Builder builder;
+    private int added;
+
+    public LimitingAttributeConsumer(int limit, Attributes.Builder builder) {
+      this.limit = limit;
+      this.builder = builder;
+    }
+
+    @Override
+    public void consume(String key, AttributeValue value) {
+      if (added++ < limit) {
```

Comment:
This works because we don't expect any attributes to have more than `Integer.MAX_VALUE / 2` size (where added will become negative and elements start being appended again), but even so it's probably clearer keep the logic precise instead of the increment shortcut. That said, this might go away soon anyways so no big deal just FYI.

```
if (added < limit) {
  buileder.set(...)
  added++;
}

Reply:
Are you suggesting the change for a functional reason, or just for clarity? what you suggest is identical functionally. 

I don't understand the Integer.MAX_VALUE comment here. Attributes limits should never come anywhere close to that, so I have zero concern around overflowing the int. Do you have a use-case that I'm not thinking of?

Reply:
Mostly just for clarity - the fact that attribute limits should never come close is true but also some hidden knowledge needed to know the code works. It's nice to reduce hidden knowledge if it's not a burden, I guess one extra line of code isn't so bad. Also allows the `added` variable to actually be true since the value stops being how many were added, but how many were consumed, after the limit.

Very small nits, just added them for future reference.

Reply:
ah, yes, I do agree with all of this. I'll be sure to include the cleanup in a future PR